src/middlewared/middlewared/etc_files/grub.py

```diff
@@ -1,17 +1,22 @@
 import subprocess
 
-from middlewared.utils import run
+from middlewared.service import CallError
+from middlewared.utils.path import module_from_file
 
 
-async def render(service, middleware):
-    try:
-        await run(["truenas-grub.py"], encoding="utf-8", errors="ignore")
-    except subprocess.CalledProcessError as e:
-        middleware.logger.error("truenas-grub.py error:\n%s", e.stderr)
-        raise
+def render(service, middleware):
+    tn_grub = module_from_file('truenas-grub', '/usr/local/bin/truenas-grub.py')
+    config = tn_grub.retrieve_grub_config()
+    kdump_enabled = middleware.call_sync('system.advanced.config')['kdump_enabled']
+    if kdump_enabled:
+        config['GRUB_CMDLINE_LINUX'].append(
```

Comment:
Why can't we do this in `truenas-grub` script?

Reply:
Because `truenas-grub` is executed during installation of scale as well and we don't have middleware running at that point to calculate crash kernel memory.

Reply:
What's so complicated in the `required_crash_kernel_memory` that it has to be a middleware method?